src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -2817,6 +2819,16 @@ _mongoc_uri_set_option_as_int64_with_error (mongoc_uri_t *uri,
 
    option = mongoc_uri_canonicalize_option (option_orig);
 
+   /* timeoutMS may not be a negative number. */
+   if (!bson_strcasecmp (option, MONGOC_URI_TIMEOUTMS) && value < 0 && value != -1) {
```

Comment:
Why is `-1` considered valid? If it is a sentinel value (perhaps meaning `unset`) I suggest defining it as a constant.

Reply:
Done, it should not accept `-1`.